typing/typeclass.ml

```diff
@@ -1931,7 +1931,20 @@ let () =
 let rec check_recmod_class_type env cty =
   match cty.pcty_desc with
   | Pcty_constr(lid, _) ->
-      ignore (Env.lookup_cltype ~use:false ~loc:lid.loc lid.txt env)
+      begin try
+        ignore (Env.lookup_cltype ~use:false ~loc:lid.loc lid.txt env)
+      with
+      | Env.Error
+          (Lookup_error
+             (location, env,
+              Illegal_reference_to_recursive_module { container; unbound; })) ->
+          raise (Env.Error
+                   (Lookup_error (location, env,
+                                  Illegal_reference_to_recursive_class_type
+                                    { container;
+                                      unbound;
+                                      class_type = lid.txt })))
+      end
```

Comment:
This is obviously not ideal, but I wasn't sure how to transform the lookup error without resorting to this